src/components/part/CreatePartDialog.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  Radio,
+  RadioGroup,
+  TextField,
+} from "@material-ui/core";
+import { FileMetadataData } from "@vertexvis/api-client-node";
+import React from "react";
+
+import { GetRes } from "../../lib/api";
+import { toFilePage } from "../../lib/files";
+import { File } from "../../lib/files";
+import { CreatePartBody, CreatePartRes } from "../../pages/api/parts";
+
+interface CreatePartDialogProps {
+  readonly open: boolean;
+  readonly onClose: () => void;
+  readonly onPartCreated: (queuedTranslationId: string) => void;
+}
+
+export default function CreatePartDialog({
+  open,
+  onClose,
+  onPartCreated,
+}: CreatePartDialogProps): JSX.Element {
+  const [file, setFile] = React.useState<string | undefined>();
+  const [files, setFiles] = React.useState<File[] | undefined>();
+  const [suppliedId, setSuppliedId] = React.useState<string | undefined>();
+  const [suppliedRevisionId, setSuppliedRevisionId] = React.useState<
+    string | undefined
+  >();
+  const [submitDisabled, setSubmitDisabled] = React.useState<boolean>(true);
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const result = await fetch(`/api/files?pageSize=25`).then((r) =>
```

Comment:
Any reason not to use SWR here? Since we've abstracted `useFiles` out in `FileTable`, using that hook in here too may lead to perf improvements